src/Uno.Extensions.Configuration/WritableOptions.cs

```diff
@@ -61,7 +63,31 @@ public async Task UpdateAsync(Func<T, T> applyChanges)
 		{
 			Directory.CreateDirectory(dir);
 		}
-		File.WriteAllText(physicalPath, json);
+		var written = false;
+		var attempt = 0;
+		while (!written && attempt++ < MaxWriteRetries)
+		{
+			await Reloader.ReadWriteLock.WaitAsync();
+			try
+			{
+				written = true;
+				File.WriteAllText(physicalPath, json);
+			}
+			catch(IOException)
+			{
+				// Only retry on IOExceptions eg sharing violation
+				written = false;
+			}
+			finally
+			{
+				Reloader.ReadWriteLock.Release();
+			}
+			if(!written)
+			{
+				await Task.Delay(50);
+			}
+		}
```

Comment:
The code need to re-read the content of the file and always reapply the update on the newest version of the file instead of trying to always rewrite the same value, which is potentially obsoleted.

Reply:
I think give we have locks around the write operation, that we can remove the multiple retries and just allow the exception to bubble up and be caught, rather than attempt to work out which write operation is correct

Reply:
I agree the lock is useless and this should never fail on mobile applications.  But on "real" platforms (Mac, Windows, Linux...) it could lead to pretty serious concurrency issues and losing information.

This is not ACID compliant and warning should be added in the documentation about this.

Reply:
Updated core removes multiple retries but leaves the locks - if the read/write then fails, this bubbles up an exception as you would expect